### PR TITLE
try par_iter on concat source

### PR DESCRIPTION
**Pull request by ahabhgk**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ dashmap = "5"
 substring = "1"
 smallvec = "1.10.0"
 memchr = "2.5.0"
+rayon = "1.6.1"
```
Comment (h-a-n-a) [inline Cargo.toml:27]: A separate feature is recommended for this case.

Comment (h-a-n-a) [reply Cargo.toml:27]: e.g `feature = "parallel"`

**Comment by Boshen** [comment]
Don't do this, this is not cache friendly, can you tell me what the actual problem is?

**Comment by ahabhgk** [comment]
> Don't do this, this is not cache friendly, can you tell me what the actual problem is?

I agree with you, and it basically have no improvement, we are facing `source.buffer` too slow. Might be helpful for more context: [switch to memfs](https://github.com/modern-js-dev/rspack/pull/1936/files#diff-3fbb9f7dfbadceab7b0c89038c54de9851f25ae957a1f91aea05b0e46da2b209R218-R237).

**Pull request closed without merging**